src/rootAstNodesBuilder.ts

```diff
@@ -892,126 +1050,117 @@ function getArguments(tokens, stringArgumentPlaceholders, argumentValidator, val
  * @returns An argument value definition.
  */
-function getArgumentDefinition(token, stringArgumentPlaceholders) {
+function getArgumentDefinition(token: string, stringArgumentPlaceholders: Placeholders): AnyArgument {
     // Check whether the token represents a null value.
     if (token === "null") {
         return {
             value: null,
-            type: "null",
-            toString: function () {
-                return this.value;
-            }
-        };
+            type: "null"
+        } as NullArgument;
     }
 
     // Check whether the token represents a boolean value.
     if (token === "true" || token === "false") {
         return {
             value: token === "true",
-            type: "boolean",
-            toString: function () {
-                return this.value;
-            }
-        };
+            type: "boolean"
+        } as BooleanArgument;
     }
 
     // Check whether the token represents a number value.
-    if (!isNaN(token)) {
+    // TODO: Relies on broken isNaN - see MDN.
+    // if (!Number.isNaN(token)) {
+    if (!isNaN(token as any)) {
         return {
-            value: parseFloat(token, 10),
-            isInteger: parseFloat(token, 10) === parseInt(token, 10),
-            type: "number",
-            toString: function () {
-                return this.value;
-            }
-        };
+            value: parseFloat(token),
+            isInteger: parseFloat(token) === parseInt(token, 10),
+            type: "number"
+        } as NumberArgument;
     }
 
     // Check whether the token is a placeholder (e.g. @@0@@) representing a string literal.
     if (token.match(/^@@\d+@@$/g)) {
         return {
             value: stringArgumentPlaceholders[token].replace('\\"', '"'),
-            type: "string",
-            toString: function () {
-                return '"' + this.value + '"';
-            }
-        };
+            type: "string"
+        } as StringPlaceholderArgument;
     }
 
     // The only remaining option is that the argument value is an identifier.
     return {
         value: token,
-        type: "identifier",
-        toString: function () {
-            return this.value;
-        }
-    };
+        type: "identifier"
+    } as IdentifierArgument;
 }
 
 /**
- * Pull any decorators off of the token stack.
+ * Pull any attributes off of the token stack.
  * @param tokens The array of remaining tokens.
  * @param stringArgumentPlaceholders The mapping of string literal node argument placeholders to original values.
- * @returns An array od decorators defined by any directly following tokens.
+ * @returns An array of attributes defined by any directly following tokens.
  */
-function getDecorators(tokens, stringArgumentPlaceholders) {
-    // Create an array to hold any decorators found.
-    const decorators = [];
+function getAttributes(tokens: string[], stringArgumentPlaceholders: Placeholders) {
```

Comment:
There's a lot of `decorators` => `attributes` - it had proliferated into quite a few places.